lib/util/viewer.js

```diff
@@ -103,6 +103,11 @@ export function extractHeadsignFromPattern(pattern, routeShortName = null) {
     headsign = pattern?.desc || ''
   }
 
+  // If we still don't have a headsign, maybe the long name will work?
+  if (isBlank(headsign)) {
+    headsign = pattern?.route?.longName || ''
+  }
```

Comment:
What happens if this is blank? Does something crash?

Reply:
Works fine!